src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests.cs

```diff
@@ -4357,11 +4360,17 @@ public class B
     }
 }
 ";
-            var compilation = CreateStandardCompilation(source);
+            var compilation = CreateStandardCompilation(source, parseOptions: TestOptions.Regular7_1);
             compilation.VerifyDiagnostics(
                 // (7,27): error CS1016: Named attribute argument expected
                 //     [A(3, z: 5, X = 6, y: 1)]
                 Diagnostic(ErrorCode.ERR_NamedArgumentExpected, "1").WithLocation(7, 27),
+                // (8,17): error CS1738: Named argument specifications must appear after all fixed arguments have been specified. Please use language version 7.2 or greater to allow non-trailing named arguments.
+                //     [A(3, z: 5, 1)]
+                Diagnostic(ErrorCode.ERR_NamedArgumentSpecificationBeforeFixedArgument, "1").WithArguments("7.2").WithLocation(8, 17),
+                // (8,11): error CS8321: Named argument 'z' is used out-of-position but is followed by an unnamed argument
+                //     [A(3, z: 5, 1)]
+                Diagnostic(ErrorCode.ERR_BadNonTrailingNamedArgument, "z").WithArguments("z").WithLocation(8, 11),
```

Comment:
@cston This is an example of cascading diagnostic that is confusing. I'll stop by.